File: assignment STL use/longest_consequitive_subsequence.java
    import java.util.ArrayList;
    import java.util.*;
    public class solution {
     
    	public static ArrayList<Integer> longestSubsequence(int[] arr){
    		// Write your code here
     
          ArrayList<Integer> list = new ArrayList<>();
     
            int maxlength = 0;
            int start = 0,end=0;
            HashMap<Integer, Boolean> map = new HashMap<>();
            for (int i = 0; i < arr.length; i++) {
                map.put(arr[i], true);
     
            }
     
     
     
            for (int i = 0; i < arr.length; i++) {
                if (map.get(arr[i]) == true) {
                    int startc = arr[i];
                    //int startc;
                    int lengthc = 1;
                    int k = arr[i]+1;
                    int l = arr[i]-1;
                  	if (map.containsKey(l))
                      continue;
                  	//System.out.println(arr[i]);
                    while (map.containsKey(k )) {
                        lengthc++;
                        map.put(k, false);
                        k++;
                    }
                    while (map.containsKey(l)) {
                        lengthc++;
                        map.put(l, false);
                        startc = l;
                        l--;
     
                    }
                    if (lengthc > maxlength) {
                      //if (lengthc > maxlength) {
                        maxlength = lengthc;
                        start = startc;
                      	end=k;
                    }
     
                }
     
     
            }
            for (int i = start; i <end; i++) {
                list.add(i);
            }
            return list;
    	}
    }

File: assignment STL use/longest_consequitive_subsequence.cpp
#include<bits/stdc++.h>
using namespace std;
vector<int> longestSubsequence(int *arr, int n){
	// Write your code here
  
  int storage,length;
  set<int> s;
  set<int>::iterator it;
  int ans=0;
  vector<int> v;
  for(int i=0;i<n;i++)
    s.insert(arr[i]);
  for(int i=0;i<n;i++){
    if(s.find(arr[i]-1)==s.end()){
      int j=arr[i];
      while(s.find(j)!=s.end())
        j++;
      if((j-arr[i])>ans){
        ans=(j-arr[i]);
        storage=arr[i];
        length=ans;
      }
      ans=max(ans,j-arr[i]);
    }
  }
  int k=0;
  for(it=s.begin();it!=s.end();it++){
    if(*it==storage){
      for(int z=0;z<length;z++){
        v.push_back(*it++);
      }
    }
  }
return v;
}  

/*this is just main function which is required on the assignment*/
